### ISSUE TITLE
Rest arguement is mishandled when controlFlowFlattening is enabled

### ISSUE THREAD

**Issue by LuoJinghua** [javascript-obfuscator/javascript-obfuscator#920]
## Expected Behavior

Expected output:
```
function f(e, h, i, j) {
    console['log'](e, h, i, j);
}
function g(...b) {
    const c = {
        'vtpEA': function (d, ...e) {
            return d(...e);
        }
    };
    c['vtpEA'](f, ...b, 0x5);
}
const a = [
    0x1,
    0x2,
    0x3
];
g(...a, 0x4);
```

## Current Behavior

Current output:
```
function f(e, h, i, j) {
    console['log'](e, h, i, j);
}
function g(...b) {
    const c = {
        'vtpEA': function (d, ...e, h) {
            return d(...e, h);
        }
    };
    c['vtpEA'](f, ...b, 0x5);
}
const a = [
    0x1,
    0x2,
    0x3
];
g(...a, 0x4);
```

## Your Environment

* Obfuscator version used: 2.12.0
* Node version used: 14.10.1

# Stack trace

## Minimal working example that will help to reproduce issue

```
function f(a, b, c, d) {
  console.log(a, b, c, d)
}

function g(...args) {
  f(...args, 5)
}
const a = [1, 2, 3]
g(...a, 4)
```

**Comment by sanex3339** [comment]
Hi. Will try to fix it today.

**Comment by sanex3339** [comment]
So, seems we can always suppose that if we have a rest argument in the start/middle of call arguments then in the function parameters this argument will always be used as the last parameter.

**Comment by sanex3339** [comment]
PR:
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/921

**Comment by sanex3339** [comment]
Released as `2.13.0`, please try it.
